tests/installation/grub_test.pm

```diff
@@ -0,0 +1,35 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use strict;
+use base "y2logsstep";
+use testapi;
+use bmwqemu ();
+
+sub run() {
+    up_and_down();
+    assert_screen "grub2", 30;
+    send_key "esc";
+    ## boot_to_snapshot is tested in other file. only xen need testing.
+    if (get_var("XEN")) {
+        send_key_until_needlematch("bootmenu-xen-kernel", 'down', 10, 5);
+    }
+    send_key "ret";
+
+}
+
+sub up_and_down() {
+    for (my $i = 0; $i <= 10; $i++) {
```

Comment:
the whole sub should not be necessary but if you would keep it: "why 10?" research about magic numbers, e.g. here http://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad